services/task/task.py

```diff
@@ -147,17 +148,36 @@ def upload(
                 debug=debug,
                 rebuild=rebuild,
             ),
+            immutable=immutable,
+        )
+
+    def upload(
+        self,
+        repoid,
+        commitid,
+        report_code=None,
+        countdown=0,
+        debug=False,
+        rebuild=False,
+    ):
+        return self.upload_signature(
+            repoid, commitid, report_code=report_code, debug=debug, rebuild=rebuild
         ).apply_async(countdown=countdown)
 
-    def notify(self, repoid, commitid, current_yaml=None, empty_upload=None):
-        self._create_signature(
+    def notify_signature(self, repoid, commitid, current_yaml=None, empty_upload=None):
+        return self._create_signature(
             "app.tasks.notify.Notify",
             kwargs=dict(
                 repoid=repoid,
                 commitid=commitid,
                 current_yaml=current_yaml,
                 empty_upload=empty_upload,
             ),
+        )
+
+    def notify(self, repoid, commitid, current_yaml=None, empty_upload=None):
```

Comment:
since now we're using different kinds of notifications here, can we rename the empty_upload to be something more descriptive? like notification_type for example?

Reply:
Yeah I think that would make sense.  I can update it in the worker too but will have to support both `empty_upload` and `notification_type` for a little bit so as to preserve backward compatibility.